utils/sync-peer-dependencies.ts

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env -S node -r @swc-node/register
+/**
+ * A script that syncs the version of all Wonder Blocks and Wonder Stuff
+ * packages in this repo using the given `package.json` file as a reference.
+ * This helps with the task of keeping Perseus' peer dependencies in line with
+ * the webapp host application.
+ */
+
+import {execSync} from "child_process";
+import * as fs from "fs";
+
+function printHelp() {
+    console.log("--- Package Dependency Sync ---");
+
+    console.log(`
+    This tool synchronizes all dev and peer dependencies found in any
+    package.json file in this repo to match the versions listed in the provided
+    package.json file. Typically, this would be the package.json file from the
+    hosting web application (ie. webapp).
+`);
+    console.log("usage: sync-peer-dependencies <package.json>");
+}
+
+// Package version regexes that we don't want to sync in
+const RestrictedPackageVersions = [
+    /^link:.*/,
+    // We don't use hot-loader/react-dom in Perseus. It's basically
+    // identical to 'react-dom' with just a few patches for webpack and we
+    // don't use webpack.
+    /hot-loader\/react-dom/,
+];
+
+// Package names that we don't want to sync in
+const RestrictedPackageNames = ["typescript"];
+
+// There are some packages and version number constructs that we don't want to
+// bring into Perseus. This function filters out packages by name or version
+// that we can't use locally.
+function filterUnusableTargetVersions(
+    targetVersions: Record<string, string>,
+    packagesInThisRepo: ReadonlyArray<string>,
+): Record<string, string> {
+    return Object.fromEntries(
+        Object.entries(targetVersions).filter(
+            ([pkgName, pkgVersion]) =>
+                !RestrictedPackageVersions.some((r) => r.test(pkgVersion)) &&
+                !RestrictedPackageNames.some((name) => name === pkgName) &&
+                !packagesInThisRepo.some((name) => name === pkgName),
```

Comment:
Questions: So the last line filters out packages in the target versions object that are not in Perseus, right? The middle line filters out packages that are restricted, while the first line filters out package versions we can't use. Does the first line need to do a regex search because the package versions sometimes have characters like `^`?

Reply:
Yes. I added more comments here, but the two `RestrictedXyz` constants have comments on them explaining what they are for.

Basically, some package versions we don't/can't use, such as `link://...` versions in webapp which are vendored-in packages in the webapp tree. We use a similar package in Perseus but can't copy in those packages verbatim. In practice, those dependencies basically don't change so we're safe there. 

Next, some packages we don't want to sync because they are things like tool (Typescript). We maintain our own version in Perseus and upgrade when we need/want to. The packages we build have types that are compatible with webapp. 

The last filter removes any package that is in this repo, `changeset` manages these cross-repo package dependencies for us so we don't want to "pull in" the versions from webapp.